examples/ConfigOnSwitchFS/ConfigOnSwitchFS.ino

```diff
@@ -0,0 +1,255 @@
+    //Local intialization. Once its business is done, there is no need to keep it around
+
+    // Extra parameters to be configured
+    // After connecting, parameter.getValue() will get you the configured value
+    // Format: <ID> <Placeholder text> <default value> <length> <custom HTML> <label placement>
+
+    // Thingspeak API Key - this is a straight forward string parameter
+    WiFiManagerParameter p_thingspeakApiKey("thingspeakapikey", "Thingspeak API Key", thingspeakApiKey, 17);
+
+    // DHT-22 sensor present or not - bool parameter visualized using checkbox, so couple of things to note
+    // - value is always 'T' for true. When the HTML form is submitted this is the value that will be 
+    //   sent as a parameter. When unchecked, nothing will be sent by the HTML standard.
+    // - customhtml must be 'type="checkbox"' for obvious reasons. When the default is checked
+    //   append 'checked' too
+    // - labelplacement parameter is WFM_LABEL_AFTER for checkboxes as label has to be placed after the input field
+
+    char customhtml[24] = "type=\"checkbox\"";
+    if (sensorDht22) {
+      strcat(customhtml, " checked");
+    }
+    WiFiManagerParameter p_sensorDht22("sensordht22", "DHT-22 Sensor", "T", 2, customhtml, WFM_LABEL_AFTER);
+
+    // I2C SCL and SDA parameters are integers so we need to convert them to char array but
+    // no other special considerations
+    char convertedValue[3];
+    sprintf(convertedValue, "%d", pinSda);
+    WiFiManagerParameter p_pinSda("pinsda", "I2C SDA pin", convertedValue, 3);
+    sprintf(convertedValue, "%d", pinScl);
+    WiFiManagerParameter p_pinScl("pinscl", "I2C SCL pin", convertedValue, 3);
+
+    // Just a quick hint
+    WiFiManagerParameter p_hint("<small>*Hint: if you want to reuse the already saved WiFi credentials, leave SSID and Password fields empty</small>");
+    
+    // Initialize WiFIManager
+    WiFiManager wifiManager;
+    
+    //add all parameters here
+    
+    wifiManager.addParameter(&p_hint);
+    wifiManager.addParameter(&p_thingspeakApiKey);
+    wifiManager.addParameter(&p_sensorDht22);
+    wifiManager.addParameter(&p_pinSda);
+    wifiManager.addParameter(&p_pinScl);  
+
+    // Sets timeout in seconds until configuration portal gets turned off.
+    // If not specified device will remain in configuration mode until
+    // switched off via webserver or device is restarted.
+    // wifiManager.setConfigPortalTimeout(600);
+
+    // It starts an access point 
+    // and goes into a blocking loop awaiting configuration.
+    // Once the user leaves the portal with the exit button
+    // processing will continue
+    if (!wifiManager.startConfigPortal()) {
+      Serial.println("Not connected to WiFi but continuing anyway.");
+    } else {
+      // If you get here you have connected to the WiFi
+      Serial.println("Connected...yeey :)");
+
+      // Getting parameters from the web form and overriding local variables' values accordingly
```

Comment:
The following lines should be moved to after the else (i.e just before line 166) so that the writeConfigFile happens even if the WiFi does not successfully connect. 

One usage scenario is that you are configuring a device to be operated somewhere else and this requires that configuration values are saved even if you have not connected to WiFi.